tool/gravity/cli/operation.go

```diff
@@ -88,48 +92,57 @@ func completeOperationPlan(localEnv, updateEnv, joinEnv *localenv.LocalEnvironme
 	}
 	switch op.Type {
 	case ops.OperationInstall:
-		return completeInstallPlan(localEnv)
+		// There's only one install operation
+		return completeInstallPlan(localEnv, op)
 	case ops.OperationExpand:
-		return completeJoinPlan(localEnv, joinEnv)
+		return completeJoinPlan(localEnv, joinEnv, op)
 	case ops.OperationUpdate:
-		return completeUpdatePlan(localEnv, updateEnv)
+		return completeUpdatePlan(localEnv, updateEnv, *op)
 	case ops.OperationUpdateEnvars:
-		return completeEnvarsPlan(localEnv, updateEnv)
+		return completeEnvarsPlan(localEnv, updateEnv, *op)
 	default:
 		return trace.BadParameter("operation type %q does not support plan completion", op.Type)
 	}
 }
 
 func getLastOperation(localEnv, updateEnv, joinEnv *localenv.LocalEnvironment, operationID string) (*ops.SiteOperation, error) {
-	operations := getBackendOperations(localEnv, updateEnv, joinEnv, operationID)
+	operations, err := getBackendOperations(localEnv, updateEnv, joinEnv, operationID)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	log.WithField("operations", oplist(operations).String()).Info("Fetched backend operations.")
 	if len(operations) == 0 {
 		if operationID != "" {
 			return nil, trace.NotFound("no operation with ID %v found", operationID)
 		}
 		return nil, trace.NotFound("no operation found")
 	}
-	op, err := getActiveOperationFromList(operations)
-	if err != nil {
-		log.WithError(err).Warn("Failed to find active operation, will fall back to last completed.")
-	}
-	if op == nil {
-		if len(operations) != 1 {
-			return nil, trace.BadParameter("multiple operations found: \n%v\n, please specify operation with --operation-id",
-				formatOperations(operations))
-		}
-		op = &operations[0]
+	if len(operations) == 1 && operationID != "" {
+		log.WithField("operation", operations[0]).Info("Fetched an operation by ID.")
+		return &operations[0], nil
 	}
-	return op, nil
+	log.Infof("Multiple operations found: \n%v\n, please specify operation with --operation-id.\n"+
```

Comment:
I did not want this as the primary source of information about multiple operations.
I think better UX would be something else rather than `plan` warning about multiple operations.
It was already suggested elsewhere that `gravity status --list-operations` would probably be more appropriate.